pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -60,29 +72,57 @@ public PartialUpsertHandler(Schema schema, Map<String, UpsertConfig.Strategy> pa
    * For example, overwrite merger will only override the prev value if the new value is not null.
    * Null values will override existing values if not configured. They can be ignored by using ignoreMerger.
    *
-   * @param previousRecord the last derived full record during ingestion.
+   * @param indexSegment the segment of the last derived full record during ingestion.
+   * @param docId the docId of the last derived full record during ingestion in the segment.
    * @param newRecord the new consumed record.
-   * @return a new row after merge
    */
-  public GenericRow merge(GenericRow previousRecord, GenericRow newRecord) {
-    for (String column : previousRecord.getFieldToValueMap().keySet()) {
+  public void merge(IndexSegment indexSegment, int docId, GenericRow newRecord) {
+    for (String column: newRecord.getFieldToValueMap().keySet()) {
```

Comment:
I think we should still go over the fields within the segment (schema) instead of the new record